components/eam/src/chemistry/yaml/aero_model/modal_aero_bcscavcoef_get_input_ts_355.yaml

```diff
@@ -0,0 +1,11 @@
+# Data at [lat=  -13.2292708024991      , lon=   139.817062054960      , k=          53 , time step=         355 ] 
+mam4xx:
+  function: modal_aero_bcscavcoef_get
+input:
+  fixed:
+    dt:    0.00
+    imode: [         1]
+    ncol: [         4]
+    isprx: [         T]
```

Comment:
Could you please change this T to true in `isprx `?

Reply:
T or true does not work in Skywalker, we need an integer, i.e.,  `T -> 1` or `F-> 0`

Reply:
currently I am using a naive way to change it (see below) as `int(isprx)` doesn't work (not sure why). we may want to change the write_var subroutine if Skywalker doesn't read logical vars. @singhbalwinder any comments?


`      integer :: logic_int 
       if (isprx(yaml%col_print, yaml%lev_print)) then
                logic_int = 1
        else
                logic_int = 0
        endif
        call write_var(unit_input, unit_output, 'isprx', logic_int)`

Reply:
If you can, would you please try changing the `write` statement in `components/eam/src/chemistry/utils/yaml_input_file_io.F90` file to print the logical like an integer (replace `L` with `I`):

```
write(unit_input,'(3A,I10,A)')'    ',trim(adjustl(fld_name)),': [', field,']'
```

This will change F->0 but the T can be a -1 or 1. We can change the above statement to print positive 1 using `abs`:
```
write(unit_input,'(3A,I10,A)')'    ',trim(adjustl(fld_name)),': [', abs(field),']'
```

You can make this change in all the subroutines ending with `_logical`.

Reply:
`abs` might not work as it is a logical, not integer. There is another Fortran intrinsic called `merge` that can be used here like:
```
write(unit_input,'(3A,I10,A)')'    ',trim(adjustl(fld_name)),': [', merge(1,0,field),']'
```

[Here ](https://gcc.gnu.org/onlinedocs/gcc-4.4.7/gfortran/MERGE.html)is some explanation of the merge function. I haven't tried it but I hope it works otherwise, we can use an explicit `if` condition.

Reply:
revised as suggested